src/servers/kfserving_http_server.cc

```diff
@@ -0,0 +1,949 @@
+{
+  if (req->method != htp_method_GET) {
+    evhtp_send_reply(req, EVHTP_RES_METHNALLOWED);
+    return;
+  }
+
+  if (model_name.empty()) {
+    evhtp_send_reply(req, EVHTP_RES_BADREQ);
+    return;
+  }
+
+  TRTSERVER_Protobuf* server_status_protobuf = nullptr;
+  TRTSERVER_Error* err = TRTSERVER_ServerModelStatus(
+      server_.get(), model_name.c_str(), &server_status_protobuf);
+  if (err == nullptr) {
+    const char* status_buffer;
+    size_t status_byte_size;
+    err = TRTSERVER_ProtobufSerialize(
+        server_status_protobuf, &status_buffer, &status_byte_size);
+    if (err == nullptr) {
+      // Request text or binary format for status?
+      std::string format;
+      const char* format_c_str = evhtp_kv_find(req->uri->query, "format");
+      if (format_c_str != NULL) {
+        format = std::string(format_c_str);
+      } else {
+        format = "text";
+      }
+
+      if (format == "binary") {
+        evbuffer_add(req->buffer_out, status_buffer, status_byte_size);
+        evhtp_headers_add_header(
+            req->headers_out,
+            evhtp_header_new("Content-Type", "application/octet-stream", 1, 1));
+      } else {
+        ServerStatus server_status;
+        if (!server_status.ParseFromArray(status_buffer, status_byte_size)) {
+          err = TRTSERVER_ErrorNew(
+              TRTSERVER_ERROR_UNKNOWN, "failed to parse server status");
+        } else {
+          std::string server_status_str = server_status.DebugString();
+          evbuffer_add(
+              req->buffer_out, server_status_str.c_str(),
+              server_status_str.size());
+        }
+      }
+    }
+  }
+
+  TRTSERVER_ProtobufDelete(server_status_protobuf);
+
+  RequestStatus request_status;
+  RequestStatusUtil::Create(
+      &request_status, err, RequestStatusUtil::NextUniqueRequestId(),
+      server_id_);
+
+  evhtp_headers_add_header(
+      req->headers_out,
+      evhtp_header_new(
+          kStatusHTTPHeader, request_status.ShortDebugString().c_str(), 1, 1));
+
+  evhtp_send_reply(
+      req, (request_status.code() == RequestStatusCode::SUCCESS)
+               ? EVHTP_RES_OK
+               : EVHTP_RES_BADREQ);
+
+  TRTSERVER_ErrorDelete(err);
+}
+
+TRTSERVER_Error*
+KFServingHTTPAPIServer::EVBufferToInput(
+    const std::string& model_name, const InferRequestHeader& request_header,
+    evbuffer* input_buffer,
+    TRTSERVER_InferenceRequestProvider* request_provider,
+    std::unordered_map<
+        std::string,
+        std::tuple<const void*, size_t, TRTSERVER_Memory_Type, int64_t>>&
+        output_shm_map)
+{
+  // Extract input data from HTTP body and register in
+  // 'request_provider'.
+  size_t buffer_length = evbuffer_get_length(input_buffer);
+  char* request_buffer = (char*)malloc(sizeof(char) * buffer_length);
+  memset(request_buffer, 0, buffer_length);
+  evbuffer_copyout(input_buffer, request_buffer, buffer_length);
+
+  // TODO MessageToJsonString to convert message to string (bytes encoded in
```

Comment:
Why is TODO still here? Is there something else needed?

Reply:
Removing that, added for the client side changes